core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/TripleStore.java

```diff
@@ -884,6 +886,19 @@ public boolean storeTriple(long subj, long pred, long obj, long context, boolean
 
 			if (stAdded) {
 				for (int i = 1; i < indexes.size(); i++) {
+					if (recordCache == null) {
+						if (requiresResize()) {
+							// map is full, resize required
```

Comment:
Did this ever happen here?

Reply:
Yeah. It did. I thought it wouldn’t, since there is a check above, but it does trigger.

Reply:
Maybe requiresResize() should just consider more space that might be needed? Using the fill level could solve this.

Reply:
I think that in this case here we might actually only require a resize because of one single operation that puts us over the 80% fill level. Could be an idea to have the requiresResize method take a priority argument, so that in this case here we can call it with low priority and then it will accept up to a fill level of 85% while if you call it with high priority it will have a fill level of 80%. Allows for some wiggle room. Or we can remove this check here and assume that a fill level of 80% should basically never cause any issues here.

Reply:
Do you have any figures about the performance impact of calling `requiresResize`? I would go for a solution where we minimize these calls to reduce the impact especially within the hot paths like triple insertion.

Reply:
I haven't run any benchmarks other than the concurrent overflow benchmark which doesn't seem very affected by this.

Reply:
Ended up removing it